packages/synapse-react-client/src/utils/functions/StringUtils.ts

```diff
@@ -16,3 +16,19 @@ export function hex2ascii(inputString: string): string {
   }
   return str
 }
+
+/**
+ * Returns a hash code from a string
+ * @param  {String} str The string to hash.
+ * @return {Number}    A 32bit integer
+ * @see https://stackoverflow.com/a/8831937
+ */
+export function hashCode(str: string) {
+  let hash = 0
+  for (let i = 0, len = str.length; i < len; i++) {
+    const chr = str.charCodeAt(i)
+    hash = (hash << 5) - hash + chr
```

Comment:
May want to include the [algorithm](https://devdocs.io/openjdk~8/java/lang/string#hashCode--) in the comments for posterity (but then again, I didn't verify that this is the actual result from this function!).
`s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]`